File: UI/fitbook_mobile/lib/models/requests/training_equipment_update_request.dart
import 'package:json_annotation/json_annotation.dart';

import '../common/api_request_body.dart';

part 'training_equipment_update_request.g.dart';

@JsonSerializable()
class TrainingEquipmentUpdateRequest implements ApiRequestBody {
  TrainingEquipmentUpdateRequest({
    required this.isRequired,
    this.note,
    required this.trainingId,
    required this.equipmentId,
  });

  final bool isRequired;
  final String? note;
  final int trainingId;
  final int equipmentId;

  factory TrainingEquipmentUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$TrainingEquipmentUpdateRequestFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$TrainingEquipmentUpdateRequestToJson(this);
}

File: FitBook.Model/Requests/TrainingEquipment/TrainingEquipmentUpdateRequest.cs
namespace FitBook.Model.Requests.TrainingEquipment;

public class TrainingEquipmentUpdateRequest
{
    public bool IsRequired { get; set; }
    public string? Note { get; set; }
    public int TrainingId { get; set; }
    public int EquipmentId { get; set; }
}
